exporters/otlp/src/otlp_environment.cc

```diff
@@ -0,0 +1,964 @@
+{
+  constexpr char kSignalEnv[]  = "OTEL_EXPORTER_OTLP_LOGS_MAX_TLS";
+  constexpr char kGenericEnv[] = "OTEL_EXPORTER_OTLP_MAX_TLS";
+
+  std::string value;
+  bool exists;
+
+  exists = GetStringDualEnvVar(kSignalEnv, kGenericEnv, value);
+  if (exists)
+  {
+    return value;
+  }
+
+  return std::string{};
+}
+
+std::string GetOtlpDefaultTracesSslTlsCipher()
+{
+  constexpr char kSignalEnv[]  = "OTEL_EXPORTER_OTLP_TRACES_CIPHER";
+  constexpr char kGenericEnv[] = "OTEL_EXPORTER_OTLP_CIPHER";
+
+  std::string value;
+  bool exists;
+
+  exists = GetStringDualEnvVar(kSignalEnv, kGenericEnv, value);
+  if (exists)
+  {
+    return value;
+  }
+
+  return std::string{};
+}
+
+std::string GetOtlpDefaultMetricsSslTlsCipher()
+{
+  constexpr char kSignalEnv[]  = "OTEL_EXPORTER_OTLP_METRICS_CIPHER";
+  constexpr char kGenericEnv[] = "OTEL_EXPORTER_OTLP_CIPHER";
+
+  std::string value;
+  bool exists;
+
+  exists = GetStringDualEnvVar(kSignalEnv, kGenericEnv, value);
+  if (exists)
+  {
+    return value;
+  }
+
+  return std::string{};
+}
+
+std::string GetOtlpDefaultLogsSslTlsCipher()
+{
+  constexpr char kSignalEnv[]  = "OTEL_EXPORTER_OTLP_LOGS_CIPHER";
+  constexpr char kGenericEnv[] = "OTEL_EXPORTER_OTLP_CIPHER";
+
+  std::string value;
+  bool exists;
+
+  exists = GetStringDualEnvVar(kSignalEnv, kGenericEnv, value);
+  if (exists)
+  {
+    return value;
+  }
+
+  return std::string{};
+}
+
+std::string GetOtlpDefaultTracesSslTlsCipherSuite()
+{
+  constexpr char kSignalEnv[]  = "OTEL_EXPORTER_OTLP_TRACES_CIPHER_SUITE";
+  constexpr char kGenericEnv[] = "OTEL_EXPORTER_OTLP_CIPHER_SUITE";
+
+  std::string value;
+  bool exists;
+
+  exists = GetStringDualEnvVar(kSignalEnv, kGenericEnv, value);
+  if (exists)
+  {
+    return value;
+  }
+
+  return std::string{};
+}
+
+std::string GetOtlpDefaultMetricsSslTlsCipherSuite()
+{
+  constexpr char kSignalEnv[]  = "OTEL_EXPORTER_OTLP_METRICS_CIPHER_SUITE";
+  constexpr char kGenericEnv[] = "OTEL_EXPORTER_OTLP_CIPHER_SUITE";
+
+  std::string value;
+  bool exists;
+
+  exists = GetStringDualEnvVar(kSignalEnv, kGenericEnv, value);
+  if (exists)
+  {
+    return value;
+  }
+
+  return std::string{};
+}
+
+std::string GetOtlpDefaultLogsSslTlsCipherSuite()
+{
+  constexpr char kSignalEnv[]  = "OTEL_EXPORTER_OTLP_LOGS_CIPHER_SUITE";
+  constexpr char kGenericEnv[] = "OTEL_EXPORTER_OTLP_CIPHER_SUITE";
+
+  std::string value;
+  bool exists;
+
+  exists = GetStringDualEnvVar(kSignalEnv, kGenericEnv, value);
+  if (exists)
+  {
+    return value;
+  }
+
+  return std::string{};
+}
+
+#endif /* ENABLE_OTLP_HTTP_SSL_TLS */
+
+std::chrono::system_clock::duration GetOtlpDefaultTracesTimeout()
+{
+  constexpr char kSignalEnv[]  = "OTEL_EXPORTER_OTLP_TRACES_TIMEOUT";
+  constexpr char kGenericEnv[] = "OTEL_EXPORTER_OTLP_TIMEOUT";
+
+  std::chrono::system_clock::duration value;
+  bool exists;
+
+  exists = GetDurationDualEnvVar(kSignalEnv, kGenericEnv, value);
+  if (exists)
+  {
+    return value;
```

Comment:
Can we check the value and make sure this value to be greater than 0?It's used by `wait_for` in OtlpHttpClient and there may be some problems when we wait for zero.

Reply:
Agreed, and fixed.